baggage/baggage.go

```diff
@@ -241,7 +242,12 @@ func NewMember(key, value string, props ...Property) (Member, error) {
 
 // NewMemberRaw returns a new Member from the passed arguments.
 //
-// The passed key must be compliant with W3C Baggage specification.
+// The passed key and value must be valid UTF-8 string.
+// However, the specific Propagators that are used to transmit baggage entries across
+// component boundaries may impose their own restrictions on baggage key.
```

Comment:
We should update `Member.String()` and `Property.String()` so that we do not encode them if its names are invalid according to W3C Baggage Propagator spec. Reference https://github.com/open-telemetry/opentelemetry-go/issues/4946#issuecomment-2018482284

Reply:
I feel we should allow `Member.String()` to return percent-encoded UTF-8 from keys, as OTel baggage allows UTF-8 in keys. If we do not encode them, it will be strange, as the result of `String` cannot pass the `Parse` method.

However, the W3C baggage propagator uses `Member.String()` directly for encoding. Maybe we should let the W3C baggage propagator drop some members if the key is UTF-8. Though additional memory allocation could occur, I am not sure whether we should go for this path or not.

Reply:
Could we provide a method to expose `baggage.List`? (it is under "go.opentelemetry.io/otel/internal/baggage")